test/integration/logic/concludable/UnifyTest.java

```diff
@@ -0,0 +1,783 @@
+/*
+ * Copyright (C) 2020 Grakn Labs
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package grakn.core.logic.concludable;
+
+import grakn.core.pattern.Conjunction;
+import grakn.core.pattern.Disjunction;
+import grakn.core.pattern.constraint.thing.HasConstraint;
+import grakn.core.pattern.constraint.thing.IsaConstraint;
+import grakn.core.pattern.constraint.thing.RelationConstraint;
+import grakn.core.pattern.constraint.thing.ValueConstraint;
+import grakn.core.pattern.variable.ThingVariable;
+import graql.lang.Graql;
+import graql.lang.pattern.variable.Reference;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static grakn.common.collection.Collections.list;
+import static grakn.common.collection.Collections.set;
+import static grakn.core.pattern.variable.VariableRegistry.createFromThings;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class UnifyTest {
```

Comment:
I think it would be more our style to call this "UnificationTest" (noun not verb)

Reply:
Agreed changed.